src/targets/registry.ts

```diff
@@ -396,78 +424,88 @@ export class RegistryTarget extends BaseTarget {
+    version: string,
+    revision: string
+  ): Promise<RegistryConfig[]> {
+    return filterAsync(this.registryConfig, async registryConfig => {
+      if (!registryConfig.linkPrereleases && isPreviewRelease(version)) {
+        this.logger.info(
+          `Preview release detected, skipping ${registryConfig.canonicalName}`
+        );
+        return false;
+      }
+
+      // If we have onlyIfPresent specified, check that we have any of matched files
+      const onlyIfPresentPattern = registryConfig.onlyIfPresent;
+      if (onlyIfPresentPattern) {
+        const artifacts = await this.artifactProvider.filterArtifactsForRevision(
+          revision,
+          {
+            includeNames: onlyIfPresentPattern,
+          }
+        );
+        if (artifacts.length === 0) {
+          this.logger.warn(
+            `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
+          );
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
   /**
    * Modifies/adds meta information regarding the package we are publishing
    */
   public async publish(version: string, revision: string): Promise<any> {
-    if (!this.registryConfig.linkPrereleases && isPreviewRelease(version)) {
-      this.logger.info('Preview release detected, skipping the target');
-      return undefined;
-    }
+    const items = await this.getValidItems(version, revision);
 
-    // If we have onlyIfPresent specified, check that we have any of matched files
-    const onlyIfPresentPattern = this.registryConfig.onlyIfPresent;
-    if (onlyIfPresentPattern) {
-      const artifacts = await this.artifactProvider.filterArtifactsForRevision(
-        revision,
-        {
-          includeNames: onlyIfPresentPattern,
-        }
-      );
-      if (artifacts.length === 0) {
-        this.logger.warn(
-          `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
-        );
-        return undefined;
-      }
+    if (items.length === 0) {
+      this.logger.warn('No suitable items found, bailing');
+      return;
     }
 
-    return this.doPublish(version, revision);
-  }
+    await withTempDir(
+      async dir => {
+        const localRepo = {
+          dir,
+          git: await this.cloneRegistry(dir),
+        };
+        await Promise.all(
+          items.map(registryConfig =>
+            this.updateVersionInRegistry(
+              registryConfig,
+              localRepo,
+              version,
+              revision
+            )
+          )
+        );
 
-  private async doPublish(version: string, revision: string) {
-    if (!RegistryTarget.localRepo) {
-      await withTempDir(
-        async dir => {
-          RegistryTarget.localRepo = {
-            dir,
-            git: await this.cloneRegistry(dir),
-          };
-          process.on('beforeExit', () => {
-            RegistryTarget.localRepo = undefined;
-            rimraf(localRepo.dir, () => {
```

Comment:
No more tempdir cleanup?

Reply:
`withTempDir` does it automatically now: https://github.com/getsentry/craft/pull/249/files#diff-c6afb4a580905af2619fdd866dc35a21d6fc15ff37364697a4355f6d2b833d4aR506 (passing `true` there means "clean up once the callback finishes)